BrainPortal/app/models/message.rb

```diff
@@ -329,4 +354,24 @@ def self.parse_markup(string) #:nodoc:
     arr.join.html_safe
   end
 
+  # before_destroy callback
+  def track_resource_usage_create
+    return true if self.message_type != "communication"  # only user to user communication is counted so far
+    CountResourceUsageForUserMessage.create(
+      :value              => 1,
```

Comment:
why not put something more useful in the value field? Like the size of the message? (sum of header, description and var_text)

Reply:
Agree, the count is the goal, but sizes could be of potential interest for reporting and ux studies. I guess we still can count message numbers. But are any downsides like any issues with it if adding message count to official report?